### PR TITLE
Expose original unmodified version string through read-only property

### PR DESCRIPTION
**Pull request by xavierdecoster**
Fixes https://github.com/NuGet/Home/issues/5068

### CODE REVIEW

**Comment by emgarten** [comment]
I'm fine with this change, but I am not sure this should ever be used over ToString(). If this is for the gallery we should normalize all SemVer 2.0.0 versions to avoid the confusion non-normalized versions cause.

**Comment by joelverhagen** [comment]
> If this is for the gallery we should normalize all SemVer 2.0.0 versions to avoid the confusion non-normalized versions cause.

We already have this behavior in the V2 feed. The `Version` property has always had the original version string. The `NormalizedVersion` has the normalized version plus any build metadata.

We improved this naming in V3. `Version` has the normalized version plus build metadata and `VerbatimVerrsion` has the original version string.

File: src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs [hunk 1]
```diff
@@ -195,5 +195,13 @@ public bool IsSemVer2
         {
             get { return (_releaseLabels != null && _releaseLabels.Length > 1) || HasMetadata; }
         }
+
+        /// <summary>
+        /// Returns the original, non-normalized version string.
+        /// </summary>
+        public string OriginalVersion
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs]: Nit: could be `public string OriginalVersion => _originalString` right? I like that syntax 😄

**Comment by emgarten** [comment]
In that case, why allow a SemVer2 package to be uploaded where the normalized and non-normalized versions are different?

I suppose the client should have thrown on these from the start and solving it here won't help. I just really want the normalized vs original issue to go away, it's painful.

**Review by emgarten — APPROVED** [review]

File: src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs (hunk 1 above)
Comment (xavierdecoster) [reply src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs]: Done https://github.com/NuGet/NuGet.Client/pull/1325/commits/4baa3cc771557a2c2ddf438cdc742d4471c92742

**Pull request merged**